homeassistant/components/switchbot/sensor.py

```diff
@@ -66,12 +77,22 @@ async def async_setup_entry(
                 entry.data[CONF_MAC],
                 entry.data[CONF_NAME],
             )
-            for sensor in coordinator.data[entry.unique_id]["data"]
+            for sensor in flatten_sensors_data(
+                coordinator.data[entry.unique_id]["data"]
+            )
             if sensor in SENSOR_TYPES
         ]
     )
 
 
+def flatten_sensors_data(sensors):
```

Comment:
Consider modifying the data structure before it's returned from the coordinator instead since we don't seem to need the original structure here in the sensor platform.

Reply:
Makes sense. :+1:  Pushed a refactor for that.